go/vt/sqlparser/sql.y

```diff
@@ -2062,6 +2076,12 @@ table_name:
     $$ = TableName{Qualifier: $1, Name: $3}
   }
 
+delete_table_name:
```

Comment:
This wouldn't allow a construct like `delete a, b.* from...`. However, I think it's ok because we've already gone above and beyond on this.